lib/help_scout/api/access_token.rb

```diff
@@ -23,29 +22,41 @@ def refresh!
         private
 
         def token_request_params
-          @token_request_params ||= {
+          @_token_request_params ||= {
             grant_type: 'client_credentials',
             client_id: HelpScout.app_id,
             client_secret: HelpScout.app_secret
           }
         end
       end
 
-      attr_reader :expires_at, :expires_in, :value
+      attr_accessor :invalid
+      attr_reader :expires_in, :value
 
       def initialize(params)
         @value = params[:access_token]
-        expires_in = params[:expires_in]
-
-        return unless expires_in
+        @expires_in = params[:expires_in]
```

Comment:
Only concern I have is a case I came across while working on caching tokens. Once we cache a new token (with an `expires_in` set), we don't really want to hydrate it's `expires_in` value when we read it from the cache because it won't be accurate. At that point, we _should_ have an `expires_at` we can use instead. Here's how I addressed this in the `token-cache` branch:

```ruby
def initialize(params)
  @value = params[:access_token]
  if params[:expires_at]
    @expires_at = DateTime.parse(params[:expires_at].to_s).to_time.utc
  elsif params[:expires_in]
    @expires_in = params[:expires_in].to_i
    @expires_at = (Time.now.utc + @expires_in)
  end
end
```
(note: also added a `require 'date'` to the top of the file)